src/sync.js

```diff
@@ -39,6 +39,13 @@ const DEBUG = true;
 // to determine the max time we're willing to wait.
 const BUFFER_PERIOD_MS = 80 * 1000;
 
+// Number of consecutive failed /sync requests
+let failedSyncCount = 0;
```

Comment:
This should be a member variable on SyncApi to match all the others (there could be multiple SyncApi objects).